homeassistant/components/neato/switch.py

```diff
@@ -20,12 +20,21 @@
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Neato switches."""
+    pass
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Neato switch with config entry."""
     dev = []
     for robot in hass.data[NEATO_ROBOTS]:
         for type_name in SWITCH_TYPES:
             dev.append(NeatoConnectedSwitch(hass, robot, type_name))
```

Comment:
We seem to be doing I/O in the switch entity init method. That's not allowed since we're in async context.

Please move all side effects out of the init method.

Reply:
@MartinHjelmare It looks like the update function of a ToggleEntity is not called after startup but some times later. Is there a way to trigger this function manually after startup?

Reply:
If we want `update` to be called during entity addition we can call `async_add_entities` with second argument `True`. If we want `update` to be called later after home assistant start event, we need to schedule this specifically with `hass.async_listen_once` from inside `async_added_to_hass` entity coroutine method.

Reply:
Yes, of course. Thank you